docs/pages/personal-data.rst

```diff
@@ -0,0 +1,75 @@
+
+The following definition comes from Article 4 of the General Data Protection Regulation (GDPR) provided by the European Union:
+
+  * ‘personal data’ means any information relating to an identified or identifiable natural person (‘data subject’); an identifiable natural person is one who can be identified, directly or indirectly, in particular by reference to an identifier such as a name, an identification number, location data, an online identifier or to one or more factors specific to the physical, physiological, genetic, mental, economic, cultural or social identity of that natural person;
+
+Event Schemas with PII
+----------------------
+
+Jupyter Telemetry requires all event schemas to define a boolean field, ``personal-data``, that explicitly states whether the event type includes personal data (as defined above). This includes any information that can be considered personally identifiable information. As stated `here <https://gdpr-info.eu/issues/personal-data/>`_:
+
+  Since the [GDPR] definition includes “any information,” one must assume that the term “personal data” should be as broadly interpreted as possible.
+
+Further, Jupyter Telemetry also requires that every property within an event has a ``categories`` field explicitly labeling the type of personal data being collected.
+
+.. code-block:: yaml
+
+    $id: example.schema
+    version: 1
+    personal-data: true
+    properties:
+      name:
+        title: Name
+        categories:
+          - unrestricted
+        description: Name of event
+        type: string
+      user:
+        title: User name
+        categories:
+           - personal-identifier
+        description: Name of the user who initiated the event.
+      affiliation:
+        title: Affiliation
+        categories:
+          - personally-identifiable-information
+        description: Affiliation of the user.
+
+
+Collecting PII
+--------------
+
+Jupyter Telemetry does **not** record any personal data by default.
+
+Any events with the field ``personal-data: true`` will not be recorded. Operators must explicitly opt-in to collect personal data by setting the ``EventLog.collect_personal_data`` attribute to ``True``.
+
+.. code-block:: python
+
+    from jupyter_telemetry import EventLog
+
+    e = EventLog(collect_personal_data=True)
+
+This will enable the EventLog to record events with ``personal-data: true``; however, these events will drop all properties that do not have ``category: unrestricted`` (``unrestricted`` is a reserved category for properties that are **always safe** to collect). Properties labeled with other categories will be removed from the event by default.
+
+To record properties with other categories, explicitly list these categories in the ``EventLog.allowed_categories`` attribute.
+
+.. code-block:: python
+
+    from jupyter_telemetry import EventLog
+
+    e = EventLog(
+      collect_personal_data=True,
+      allowed_categories=['personally-identifiable-information']
```

Comment:
Since this is set at an EventLog level, this doesn't let us tune what kinda information we want to collect from which schema. In one schema, 'personal-identifier' might be a pseudonym, and in another it could be your phone number. Those two should be treated completely differently, and keeping this at the global level doesn't allow admins to do that. Schemas could use something more specific than 'personal-identifier', but that requires co-ordination between different schema writers to make sure they don't overlap. This pseudonym / phone number mixup could happen unintentionally, for example, when you have a different spawner / authenticator combo.

Reply:
What do you think about moving this instead to [allowed_schemas](https://github.com/jupyter/telemetry/blob/master/jupyter_telemetry/eventlog.py#L45)? Instead of strings, you can optionally add a dict with schema name, allowed categories, etc.

Reply:
I like this idea a lot, @yuvipanda. Let me refactor things a bit and see what we think.

Reply:
One thing that isn't clear here—should individual *handlers* dictate what type of data is logged by that handler? Right now, `allowed_schemas` is set at the `EventLog` level. Should we, instead, set `allowed_schemas` (with extra metadata about categories) in each handler?

Reply:
I think it should be layered - handlers inherit `allowed_schemas` but can impose their own.